src/common/utils/auth_tests.rs

```diff
@@ -31,94 +51,2254 @@ mod tests {
+                bypass_check: true,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
+
+    #[tokio::test]
+    async fn get_latest_traces() {
+        test_auth(
+            Method::GET,
+            format!("api/{ORG_ID}/STREAM_NAME/traces/latest"),
+            AuthExtractor {
+                auth: AUTH_HEADER_VAL.to_string(),
+                // Are these empty strings correct?
+                method: format!(""),
+                o2_type: format!(""),
+                org_id: format!(""),
+                bypass_check: true,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
+
+    // Tests for routes defined in handler::http::metrics::ingest.
+
+    #[tokio::test]
+    async fn otlp_metrics_write() {
+        test_auth(
+            Method::POST,
+            format!("api/{ORG_ID}/v1/metrics"),
+            AuthExtractor {
+                auth: AUTH_HEADER_VAL.to_string(),
+                method: format!("{POST_METHOD}"),
+                o2_type: format!("stream:{ORG_ID}"),
+                org_id: format!("{ORG_ID}"),
+                bypass_check: true,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
+
+    #[tokio::test]
+    async fn metrics_json() {
+        test_auth(
+            Method::POST,
+            format!("api/{ORG_ID}/ingest/metrics/_json"),
+            AuthExtractor {
+                auth: AUTH_HEADER_VAL.to_string(),
+                method: format!("{POST_METHOD}"),
+                o2_type: format!("stream:{ORG_ID}"),
+                org_id: format!("{ORG_ID}"),
+                bypass_check: true,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
+
+    // Tests for routes defined in handler::http::request::promql.
+
+    #[tokio::test]
+    async fn promql_remote_write() {
+        test_auth(
+            Method::POST,
+            format!("api/{ORG_ID}/prometheus/api/v1/write"),
+            AuthExtractor {
+                auth: AUTH_HEADER_VAL.to_string(),
+                method: format!("{POST_METHOD}"),
+                o2_type: format!("stream:{ORG_ID}"),
+                org_id: format!("{ORG_ID}"),
+                bypass_check: true,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
+
+    #[tokio::test]
+    async fn promql_query_get() {
+        test_auth(
+            Method::GET,
+            format!("api/{ORG_ID}/prometheus/api/v1/query"),
+            AuthExtractor {
+                auth: AUTH_HEADER_VAL.to_string(),
+                // Should these be empty strings?
+                method: format!(""),
+                o2_type: format!(""),
+                org_id: format!(""),
+                bypass_check: true,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
```

Comment:
?

Reply:
Yes intentional. For these apis -
/_search (POST method),
/prometheus/api/v1/query
/resources
/format_query
/prometheus/api/v1/series
/traces/latest
clusters
query_manager
/short
/ws
we ignore the openfga checks in the middleware. This is why the `bypass_check` field is `true`.